packages/legend-query-builder/src/stores/filter/QueryBuilderFilterState.ts

```diff
@@ -988,25 +1163,35 @@ export class QueryBuilderFilterState
             `Derived property parameter value for ${node.condition.propertyExpressionState.title} is missing or invalid`,
           );
         }
+        if (
+          node.condition.rightConditionValue instanceof
+            FilterPropertyExpressionStateConditionValueState &&
+          !node.condition.rightConditionValue.propertyExpressionState.isValid
```

Comment:
could you please correct me if my understanding about `FilterPropertyExpressionStateConditionValueState` is wrong?
e.g. if we drag and drop a node from exploreTree to the right hand side of filter node, it will use `FilterPropertyExpressionStateConditionValueState` ?
if so, may i know why this error message is specific to `Derived property parameter`? 
Should we also add this check to `isInvalidFilterValue()` ?

Reply:
`FilterPropertyExpressionStateConditionValueState` will be used if we DND a node from explorer tree to the right hand side of the filter node OR if we DND a column from the fetch structure panel to the right hand side of the filter node. If the property expression on the right hand side of the filter node is a derived property and it's missing a parameter value, then we need to show an error for that.

I've renamed some of the functions used here to try to make them clearer.
- Renamed `isInvalidFilterValue()` to `isInvalidValueSpecFilterValue()` since it checks if the right hand side is a ValueSpecification and if it's an invalid ValueSpecification
- Added a new `isInvalidFilterPropertyExpressionState()` function to check if the left hand side is an invalid property expression state.
- Added a new `isInvalidPropertyExpressionStateFilterValue()` function to check if the right hand side is a PropertyExpressionState and if it's an invalid PropertyExpressionState.